java/src/main/java/org/eclipse/ditto/client/live/internal/LiveThingHandleImpl.java

```diff
@@ -103,40 +104,32 @@ protected LiveFeatureHandleImpl createFeatureHandle(final ThingId thingId, final
             final Class<T> type,
             final Consumer<RepliableMessage<T, U>> handler) {
 
-        argumentNotNull(registrationId, "registrationId");
-        argumentNotNull(subject, "subject");
-        argumentNotNull(type, "type");
-        argumentNotNull(handler, "handler");
-
+        LiveMessagesUtil.checkSubject(subject);
```

Comment:
@stmaute I don't understand that change .. you replaced checks:
* `registrationId != null`
* `subject != null`
* `type != null`
* `handler != null`

with only:
* `subject != null`

But those are methods invoked by client users and we must forbid passing in `null`.
So what was the intention here?

Reply:
These checks where made several times in different methods. I tried to simplified it and check the parameter in methods were they are used. E.g. the registrationId is checked [here](https://github.com/eclipse/ditto-clients/blob/1b35c8b315d0ae4d1b799d1904a36150ca73a0f4/java/src/main/java/org/eclipse/ditto/client/internal/HandlerRegistry.java#L61) and the handler is checked [here](https://github.com/bosch-io/ditto-clients/blob/f78e86a630bc549cde386960afc8e108a59381d9/java/src/main/java/org/eclipse/ditto/client/live/internal/LiveMessagesUtil.java#L196)

Added argumentNotNull for the type variable in some classes.

In my eyes it isn't necessary to check the parameters multiple times. Or do I miss out something here?

Reply:
Ah, yes .. you're right. I didn't follow the parameters to the next call stack ..
Then it's all good.